src/main/java/net/minecraftforge/event/level/LevelEvent.java

```diff
@@ -117,4 +125,78 @@ public ServerLevelData getSettings()
             return settings;
         }
     }
+
+    /**
+     * Called by NaturalSpawner to gather a list of all possible entities that can spawn at the specified location.
+     * If an entry is added to the list, it needs to be a globally unique instance.
+     * The event is called in {@link net.minecraft.world.level.NaturalSpawner#getRandomSpawnMobAt(ServerLevel,
+     * StructureManager, ChunkGenerator, MobCategory, RandomSource, BlockPos)}.
+     * Canceling the event will result in an empty list, meaning no entity will be spawned.
+     */
+    @Cancelable
+    public static class PotentialSpawns extends LevelEvent
+    {
+        private final MobCategory mobcategory;
+        private final BlockPos pos;
+        private final List<MobSpawnSettings.SpawnerData> list;
+        private final List<MobSpawnSettings.SpawnerData> view;
+
+        public PotentialSpawns(LevelAccessor level, MobCategory category, BlockPos pos, WeightedRandomList<MobSpawnSettings.SpawnerData> oldList)
+        {
+            super(level);
+            this.pos = pos;
+            this.mobcategory = category;
+            if (!oldList.isEmpty())
+                this.list = new ArrayList<>(oldList.unwrap());
+            else
+                this.list = new ArrayList<>();
+
+            this.view = Collections.unmodifiableList(list);        }
+
+        /**
+         * @return the category of the mobs in the spawn list.
+         */
+        public MobCategory getMobCategory()
+        {
+            return mobcategory;
+        }
+
+        /**
+         * @return the block position where the chosen mob will be spawned.
+         */
+        public BlockPos getPos()
+        {
+            return pos;
+        }
+
+        /**
+         * @return the list of mobs that can potentially be spawned.
```

Comment:
Use the inline `{@return}` tag instead.
```suggestion
         * {@return the list of mobs that can potentially be spawned}
```